src/components/TerraformConfig/TerraformConfig.tsx

```diff
@@ -0,0 +1,136 @@
+import { Alert, Button, Modal, useStyles2 } from '@grafana/ui';
+import { config as runtimeConfig } from '@grafana/runtime';
+import { InstanceContext } from 'contexts/InstanceContext';
+import React, { useContext, useState } from 'react';
+import { Clipboard } from 'components/Clipboard';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { TFCheckConfig, TFConfig, TFProbeConfig } from './terraformTypes';
+import { checkToTF, probeToTF, sanitizeName } from './terraformConfigUtils';
+import { checkType } from 'utils';
+import { CheckType } from 'types';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  modal: css`
+    max-height: 100%;
+  `,
+  clipboard: css`
+    max-height: 500px;
+  `,
+  text: css`
+    max-width: 600px;
+  `,
+});
+
+export const TerraformConfig = () => {
+  const { instance } = useContext(InstanceContext);
+  const [showModal, setShowModal] = useState(false);
+  const [error, setError] = useState<string | undefined>();
+  const [config, setConfig] = useState<TFConfig | undefined>();
+  const styles = useStyles2(getStyles);
+
+  const generateTFConfig = async (): Promise<TFConfig> => {
+    const checks = await instance.api?.listChecks();
+    const probes = await instance.api?.listProbes();
+    if (!checks || !probes) {
+      throw new Error("Couldn't generate TF config");
+    }
+    const checksConfig = checks?.reduce<TFCheckConfig>((acc, check) => {
+      const type = checkType(check.settings);
+      if (type === CheckType.Traceroute) {
+        return acc;
+      }
+      const checkConfig = checkToTF(check);
+      const formattedJob = sanitizeName(check.job);
+      if (!acc[formattedJob]) {
+        acc[formattedJob] = checkConfig;
+      } else {
+        throw new Error(`Cannot generate TF config for checks with duplicate job names: ${check.job}`);
+      }
+      return acc;
+    }, {});
+
+    const probesConfig = probes?.reduce<TFProbeConfig>((acc, probe) => {
+      if (probe.public) {
+        return acc;
+      }
+      const probeConfig = probeToTF(probe);
+      const sanitizedName = sanitizeName(probe.name);
+      if (!acc[sanitizedName]) {
+        acc[sanitizedName] = probeConfig;
+      } else {
+        throw new Error(`Cannot generate TF config for probes with duplicate job names: ${probe.name}`);
```

Comment:
I'm going to guess that terraform names are case insensitve?

Because probe names cannot be duplicated, unless I'm missing something...

Reply:
good point, just being extra cautious